### PR TITLE
Azure AD Oauth Login for Redash

### PR DESCRIPTION
**Pull request by georgewfisher**
## What type of PR is this? 

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] New Query Runner (Data Source) 
- [ ] New Alert Destination
- [ ] Other

## Description
Enable Azure AD Oauth login for Redash. Enables a login "with work or school account" option to be configured for the Redash login page which allows the user to authenticate using their Azure/Microsoft work or school identity and generate an associated Redash account.

Implementation is based on Google Oauth code in Redash that already exists. Some consolidation between Google and Azure Oauth code might be possible, but due to need to separate UI, I felt it was cleaner to keep them separate. I have not added RBAC (role-based access control) role verification to this PR but could easily do so (I plan to do this in a future PR).

## How is this tested?

- [X] Unit tests (pytest, jest)
- [ ] E2E Tests (Cypress)
- [X] Manually
- [ ] N/A

Manual testing scope involved local and cloud deployment test setup, test domain configuration, test login, test user account creation, verify user account is functional.

## Related Tickets & Documents

[https://github.com/getredash/website/pull/658](https://github.com/getredash/website/pull/658)

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

Azure AD login using "Microsoft work or school" customer facing langugage:
image

Authentication settings UI under general settings. Shows up only if Azure authentication is enabled.
image

### CODE REVIEW

**Comment by georgewfisher** [comment]
Looking at test failures

**Comment by georgewfisher** [comment]
@susodapop Looks like all checks are passing. What do I need to do to get this reviewed? Let me know if you need anything or have any questions!

**Comment by georgewfisher** [comment]
@susodapop Can you confirm if you are blocked on anything from me? Do I need to do anything to apply to the "Team Review"?

I can also help this process by bringing a 2nd or 3rd set of eyes to supplement your review. Let me know if this would help

**Comment by susodapop** [comment]
> I can also help this process by bringing a 2nd or 3rd set of eyes to supplement your review. Let me know if this would help

Yes absolutely this would help. I appreciate your efforts to get this across the line 🙏

File: requirements_bundles.txt [hunk 1]
```diff
@@ -4,5 +4,5 @@
 # It's automatically installed when running npm run bundle
 
 # These can be removed when upgrading to Python 3.x
-importlib-metadata>=1.6  # remove when on 3.8
+importlib-metadata>=1.6,<5.0  # remove when on 3.8
```
Comment (georgewfisher) [inline requirements_bundles.txt:7]: Brought over https://github.com/getredash/redash/pull/5831 to make tests pass

**Comment by guidopetri** [comment]
@georgewfisher thanks for your contribution! We've made quite a few changes after the switch to community-driven development, would you mind rebasing off of `master` and resolving merge conflicts so that the CI re-runs?

**Pull request closed without merging**